src/libraries/System.Private.Xml.Linq/src/System/Xml/Linq/Extensions.cs

```diff
@@ -277,8 +277,6 @@ public static IEnumerable<XElement> DescendantsAndSelf(this IEnumerable<XElement
             return DocumentOrderIterator<T>(source);
         }
 
-        // TODO-NULLABLE: Consider changing to T? instead.
-        // If we do it, we will also need to change XNodeDocumentOrderComparer to implement IComparer<XNode?> instead.
```

Comment:
comparer seem to assume everything can be nullable so wasn't sure what this was about, user can also pass nullable T

Reply:
> user can also pass nullable T

Not true, below snippet gives you this error 
```
The type 'System.Xml.Linq.XNode?' cannot be used as type parameter 'T' in the generic type or method 'Extensions.InDocumentOrder<T>(IEnumerable<T>)'. Nullability of type argument 'System.Xml.Linq.XNode?' doesn't match constraint type 'System.Xml.Linq.XNode'.
```

```cs
var source = new List<XNode?>();
IEnumerable<XNode?> t = source.InDocumentOrder();
```

Reply:
I `Comparer<T>` is really weird as it always should be nullable, I've fixed it to make this scenario work

Reply:
Nice, thanks.